pkg/extensions/extension_auth_query.go

```diff
@@ -0,0 +1,113 @@
+//go:build auth_query && !debug
+// +build auth_query,!debug
+
+package extensions
+
+import (
+	"encoding/json"
+	"errors"
+	"net"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/mux"
+
+	"zotregistry.io/zot/pkg/api/config"
+	"zotregistry.io/zot/pkg/api/constants"
+	"zotregistry.io/zot/pkg/log"
+)
+
+var ErrInvalidIP = errors.New("invalid ip")
+
+type AuthQueryResponse struct {
+	Auth AuthInfo `json:"auth"`
+}
+
+type AuthInfo struct {
+	Type         string `json:"type"`
+	BearerIssuer string `json:"bearerIssuer"`
+	Enabled      bool   `json:"enabled"`
+}
+
+func getIP(request *http.Request) (string, error) {
+	// Get IP from the X-REAL-IP header
+	ip := request.Header.Get("X-REAL-IP") //nolint: varnamelen
+
+	netIP := net.ParseIP(ip)
+	if netIP != nil {
+		return ip, nil
+	}
+
+	// Get IP from X-FORWARDED-FOR header from the right to left (to get proxy added address)
+	ips := request.Header.Get("X-FORWARDED-FOR")
+
+	splitIps := strings.Split(ips, ",")
+	for i := len(splitIps) - 1; i >= 0; i-- {
+		ip := strings.TrimSpace(splitIps[i])
+
+		// header can contain spaces too, strip those out.
+		netIP = net.ParseIP(ip)
+		if netIP != nil {
+			return ip, nil
+		}
+	}
+
+	// Get IP from RemoteAddr
+	ip, _, err := net.SplitHostPort(request.RemoteAddr)
+	if err != nil {
+		return "", err
+	}
+
+	netIP = net.ParseIP(ip)
+	if netIP != nil {
+		return ip, nil
+	}
+
+	return "", ErrInvalidIP
+}
+
+type authHandler struct {
+	config *config.Config
+	log    log.Logger
+}
+
+func (authHandler *authHandler) getInfo(response http.ResponseWriter, request *http.Request) {
+	remoteAddr, err := getIP(request)
+	if err != nil {
+		authHandler.log.Error().Err(err).Msg("auth query: couldn't obtain request ip")
+		response.WriteHeader(http.StatusInternalServerError)
+
+		return
+	}
+
+	if !strings.Contains(remoteAddr, "127.0.0.1") {
```

Comment:
This part is not correct (my bad actually)

server (remote-ip) <-> client (client-ip)

Client has to reach this server URL without authN